src/hooks/core/push-notification/usePushNotificationHook.ts

```diff
@@ -152,11 +154,21 @@ const usePushNotificationHook = () => {
     }
     if (notification?.data?.type === 'message.new') {
       if (notification?.userInteraction) {
-        const selectedChannel = await ChannelList.getSchemaById(
-          localDb,
-          notification?.data?.channel_id
+        const isSameChannel = selectedChannel?.id === notification?.data?.channel_id;
+        const isOnChatScreen = ['SignedChatScreen', 'AnonymousChatScreen'].includes(
+          currentChatScreen || ''
         );
-        setSelectedChannel(selectedChannel);
+
+        if (isSameChannel && isOnChatScreen) {
+          const channel = new ChannelList({
+            id: notification?.data?.channel_id,
+            channelType: 'PM'
+          });
+          return fetchChannelDetail(channel);
+        }
+
+        const channel = await ChannelList.getSchemaById(localDb, notification?.data?.channel_id);
+        setSelectedChannel(channel);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code has been refactored to handle incoming messages based on the current chat screen and selected channel. It seems to be logically correct, but ensure that all edge cases are handled and tested thoroughly.

```diff
+ const isSameChannel = selectedChannel?.id === notification?.data?.channel_id;
+ const isOnChatScreen = ['SignedChatScreen', 'AnonymousChatScreen'].includes(
+   currentChatScreen || ''
+ );
+ 
+ if (isSameChannel && isOnChatScreen) {
+   const channel = new ChannelList({
+     id: notification?.data?.channel_id,
+     channelType: 'PM'
+   });
+   return fetchChannelDetail(channel);
+ }
+ 
+ const channel = await ChannelList.getSchemaById(localDb, notification?.data?.channel_id);
+ setSelectedChannel(channel);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->